### PR TITLE
test: add workflow trigger by gaia

### PR DESCRIPTION
**Pull request by yaruwangway**
Closes: #XXX

Gaia uses hermes CI test to test a specific version of gaia. The test will only run when gaia tags a new release or manually trigger.

Ref:
- [The original issue in gaia repo](https://github.com/cosmos/gaia/issues/2067)
- [The corrsponding trigger yml file](https://github.com/cosmos/gaia/pull/2712) 
## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: .github/workflows/test_for_gaia.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        RUST_LOG: info
+        RUST_BACKTRACE: 1
+        NO_COLOR_LOG: 1
+        CHAIN_COMMAND_PATHS: ${{ matrix.chain.command }}
+        ACCOUNT_PREFIXES: ${{ matrix.chain.account_prefix }}
+      with:
+        max_attempts: 2
+        timeout_minutes: 90
+        command: |
+          CHAIN_COMMAND_PATHS=${{ github.workspace }}/gaia/build/gaiad ACCOUNT_PREFIXES=cosmos cargo test -p ibc-integration-test -- tests::transfer::test_ibc_transfer
```
Comment (yaruwangway) [inline .github/workflows/test_for_gaia.yml:63]: more test will be add after the setup is tested.

File: .github/workflows/test_for_gaia.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: Test Triggered by Gaia
```
Comment (romac) [inline .github/workflows/test_for_gaia.yml:1]:
```suggestion
name: Test new Gaia release
```

File: .github/workflows/test_for_gaia.yml (hunk 1 above)
Comment (romac) [inline .github/workflows/test_for_gaia.yml:63]:
```suggestion
          cargo test -p ibc-integration-test -- tests::transfer::test_ibc_transfer
```

File: .github/workflows/test_for_gaia.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        cd ../
+    - name: Make gaiad executable
+      run: chmod +x ${{ github.workspace }}/gaia/build/gaiad
+    - uses: nick-fields/retry@v2
+      env:
+        RUST_LOG: info
+        RUST_BACKTRACE: 1
+        NO_COLOR_LOG: 1
+        CHAIN_COMMAND_PATHS: ${{ matrix.chain.command }}
+        ACCOUNT_PREFIXES: ${{ matrix.chain.account_prefix }}
```
Comment (romac) [inline .github/workflows/test_for_gaia.yml:58]:
```suggestion
        CHAIN_COMMAND_PATHS: ${{ github.workspace }}/gaia/build/gaiad
        ACCOUNT_PREFIXES: cosmos
```

File: .github/workflows/test_for_gaia.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        override: true
+    - uses: Swatinem/rust-cache@v2
+    - uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: -p ibc-integration-test --no-fail-fast --no-run
+    - uses: actions/setup-go@v4
+      with:
+        go-version: 1.20.x
+    - name: Checkout hermes version
```
Comment (romac) [inline .github/workflows/test_for_gaia.yml:41]:
```suggestion
    - name: Checkout Hermes ${{ github.event.client_payload.hermes_version }}
```

File: .github/workflows/test_for_gaia.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      run: |
+        git checkout ${{ github.event.client_payload.hermes_version }}
+    - name: Fetch Gaia Repo
```
Comment (romac) [inline .github/workflows/test_for_gaia.yml:44]:
```suggestion
    - name: Build Gaia ${{ github.event.client_payload.gaiad_version }}
```

**Review by romac — CHANGES_REQUESTED** [review]

**Comment by yaruwangway** [comment]
Hi, I put this in blocked for now, we might want to do it differently. Thank you for reviewing!

**Comment by romac** [comment]
@yaruwangway I am closing this, feel free to re-open if you choose to stick with this approach.

**Comment by vehicles4real** [comment]
Thanks

**Comment by vehicles4real** [comment]
Thanks

**Pull request closed without merging**